build_resolvers/lib/src/build_asset_uri_resolver.dart

```diff
@@ -198,17 +197,20 @@ String assetPath(AssetId assetId) =>
 
 /// Returns all the directives from a Dart library that can be resolved to an
 /// [AssetId].
-Set<AssetId> _parseDirectives(String content, AssetId from) =>
-    HashSet.of(parseString(content: content, throwIfDiagnostics: false)
-        .unit
-        .directives
-        .whereType<UriBasedDirective>()
-        .where((directive) {
-          var uri = Uri.parse(directive.uri.stringValue);
-          return !_ignoredSchemes.any(uri.isScheme);
-        })
-        .map((d) => AssetId.resolve(d.uri.stringValue, from: from))
-        .where((id) => id != null));
+Set<AssetId> _parseDirectives(String content, AssetId from) => HashSet.of(
+      parseString(content: content, throwIfDiagnostics: false)
+          .unit
+          .directives
+          .whereType<UriBasedDirective>()
+          .map((directive) => directive.uri.stringValue)
+          .where((uriContent) {
```

Comment:
We could do another `.whereType<String>()` to avoid the null check in the body here and the `!` in the following map function, but either way sgtm.